### PR TITLE
Improve _dotfiles task of generator and clean up a bit

### PR DESCRIPTION
**Pull request by racse1**
Hi @mi2oon, here's the another one. I started with `_dotfiles` task, it has a bunch of repetetive code, so I decided to get rid of that stuff and now it looks like a `_directories` task.

Alongside with that I did a bit of clean up here and there: esnextifying some code, removing white spaces, sorting things, adding labels.

Please let me know if there's anything I should also look at or change 😄

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
-const superb    = require('superb');
-const mkdirp    = require('mkdirp');
 const chalk     = require('chalk');
 const Generator = require('yeoman-generator');
+const mkdirp    = require('mkdirp');
+const superb    = require('superb');
+const yosay     = require('yosay');
+
+const OPTIONS = {
+    KEEP_SILENCE: 'keep-silence',
```
Comment (racse1) [inline app/index.js]: Not sure whether you use such a naming style for defining a constants or not... It also supposed to behave a bit like a enum, so it gets complicated. Let me know if we need it at all and if you have any thouhts regarding naming.

File: app/index.js [hunk 2]
```diff
@@ -301,16 +297,14 @@ module.exports = class extends Generator {
     }
 
     install() {
-        if (!this.options['skip-install']){
+        if (!this.options[OPTIONS.SKIP_INSTALL]){
             this.spawnCommandSync('npm', ['update', '--save', '--save-exact']);
             this.spawnCommandSync('npm', ['update', '--save-dev', '--save-exact']);
         }
-
-        this.installDependencies({bower: false});
```
Comment (racse1) [inline app/index.js]: `npm update` will do stuff 😄

File: app/index.js [hunk 3]
```diff
@@ -14,47 +19,38 @@ function capitalizeKebabCase(name) {
...
     }
 
     initializing() {
-        this.option('keep-silence', {
-            desc: 'Test framework to be invoked',
-            type: Boolean
-        });
-
-
-        this.projectYear = new Date().getFullYear();
```
Comment (racse1) [inline app/index.js]: Didn't find any references to this one, so I got rid of it. Is it OK?

File: app/index.js [hunk 4]
```diff
@@ -14,47 +19,38 @@ function capitalizeKebabCase(name) {
...
-        this.option('keep-silence', {
-            desc: 'Test framework to be invoked',
+        this.answers = {};
+
+        this.option(OPTIONS.SKIP_INSTALL, {
+            desc: 'Do not install npm dependencies',
             type: Boolean
         });
 
-        this.option('babel', {
```
Comment (racse1) [inline app/index.js]: Looks like this option isn't needed anymore. Didn't find any references to it.

File: app/index.js (hunk 1 above)
Comment (mi2oon) [reply app/index.js]: I like the way, in this way they are easy to identify :)

**Comment by racse1** [comment]
> Lets bring it in.. shall we version bump? (@mi2oon)

@mi2oon I think so 👍 There's a couple of Gulp-related improvements left, but probably it makes sense to take them into a next release.

**Comment by mi2oon** [comment]
@racse1 Lets just wait a little.. new year new version ;)

**Pull request merged**